packages/back-end/src/services/safeRolloutSnapshots.ts

```diff
@@ -0,0 +1,135 @@
+import {
+  DEFAULT_METRIC_WINDOW,
+  DEFAULT_METRIC_WINDOW_DELAY_HOURS,
+  DEFAULT_METRIC_WINDOW_HOURS,
+  DEFAULT_PROPER_PRIOR_STDDEV,
+  DEFAULT_REGRESSION_ADJUSTMENT_DAYS,
+} from "shared/constants";
+import { isDefined } from "shared/util";
+import {
+  ExperimentMetricInterface,
+  getAllMetricIdsFromExperiment,
+  isBinomialMetric,
+  isFactMetric,
+} from "shared/experiments";
+import {
+  MetricForSnapshot,
+  SafeRolloutSnapshotAnalysisSettings,
+  SafeRolloutSnapshotInterface,
+} from "back-end/src/validators/safe-rollout";
+import {
+  ExperimentSnapshotAnalysisSettings,
+  ExperimentSnapshotSettings,
+} from "back-end/types/experiment-snapshot";
+
+export function getMetricForSnapshot(
+  id: string | null | undefined,
+  metricMap: Map<string, ExperimentMetricInterface>,
+  settingsForSnapshotMetrics: MetricForSnapshot[]
+): MetricForSnapshot | null {
+  if (!id) return null;
+  const metric = metricMap.get(id);
+  if (!metric) return null;
+  const metricSnapshotSettings = settingsForSnapshotMetrics?.find(
+    (s) => s.id === id
+  );
+  return {
+    id,
+    settings: {
+      datasource: metric.datasource,
+      type: isBinomialMetric(metric) ? "binomial" : "count",
+      aggregation: ("aggregation" in metric && metric.aggregation) || undefined,
+      cappingSettings: metric.cappingSettings,
+      denominator: (!isFactMetric(metric) && metric.denominator) || undefined,
+      sql: (!isFactMetric(metric) && metric.sql) || undefined,
+      userIdTypes: (!isFactMetric(metric) && metric.userIdTypes) || undefined,
+    },
+    computedSettings: {
+      windowSettings: {
+        delayValue:
+          metric.windowSettings.delayValue ?? DEFAULT_METRIC_WINDOW_DELAY_HOURS,
+        delayUnit: metric.windowSettings.delayUnit ?? "hours",
+        type: metric.windowSettings.type ?? DEFAULT_METRIC_WINDOW,
+        windowUnit: metric.windowSettings.windowUnit ?? "hours",
+        windowValue:
+          metric.windowSettings.windowValue ?? DEFAULT_METRIC_WINDOW_HOURS,
+      },
+      properPrior: false,
+      properPriorMean: 0,
+      properPriorStdDev: DEFAULT_PROPER_PRIOR_STDDEV,
+      regressionAdjustmentDays: DEFAULT_REGRESSION_ADJUSTMENT_DAYS,
+      regressionAdjustmentEnabled: false,
```

Comment:
This should be enabled, or it should be whatever their org default is. I'd lean towards on if possible for sure. Probably would just mimic the regular flow for this, except we won't have a killswitch on safe rollout page.